src/platform/datapath_raw_socket.c

```diff
@@ -105,7 +173,11 @@ CxPlatTryAddSocket(
     if (Success) {
         CxPlatHashtableInsert(&Pool->Sockets, &Socket->Entry, Socket->LocalAddress.Ipv4.sin_port, &Context);
     }
+
     CxPlatRwLockReleaseExclusive(&Pool->Lock);
+
+Error:
```

Comment:
We need to clean up the aux socket on failure. `CxPlatRemoveSocket` isn't called on failure.